ReLU6.lua

```diff
@@ -1,7 +1,9 @@
-local ReLU6, parent = torch.class('nn.ReLU6', 'nn.Module')
+local ReLU6, parent = torch.class('nn.ReLU6', 'nn.HardTanh')
```

Comment:
while this is the most effective approach (and i think so as well), after discussing with @jonathantompson , he was concerned that having a self.max_val in ReLU6 will make people believe that it's a tunable parameter.
Can you instead not subclass it but call the THNN.HardTanh.updateOutput functions with the value 6 directly?